### DISCUSSION TITLE
Axis names for 2-axis lathe

### DISCUSSION THREAD

**Discussion by mvackel** [Paciente8159/uCNC#505] [General]
When configuring uCNC for 2-axis, the axis get names "X" and "Y". Is there a way to change to "X" and "Z" as is the usual convention for a lathe?

**Comment by Paciente8159** [comment]
There is a way to change axis indexes but it as not been tested. I'll try to test to see if it works on the emulator

**Comment by Paciente8159** [comment]
[Here is the place](https://github.com/Paciente8159/uCNC/blob/2ddfae347897f6f950f96a1a3a4c8e3d471484dc/uCNC/src/hal/kinematics/kinematicdefs.h#L44)

If you replace this by:

```
#if (AXIS_COUNT > 0)
#define AXIS_X 0
#endif
//swap Y and Z Axis
#if (AXIS_COUNT > 2)
#define AXIS_Y 2
#endif
#if (AXIS_COUNT > 1)
#define AXIS_Z 1
#endif
#if (AXIS_COUNT > 3)
#define AXIS_A 3
#endif
#if (AXIS_COUNT > 4)
#define AXIS_B 4
#endif
#if (AXIS_COUNT > 5)
#define AXIS_C 5
#endif
```
..works but there is at least a caveat.

Note that by doing this you will also be reporting coordinates like <Pos: X,Z,Y>

**Reply by mvackel** [reply]
This code seems to change the axes, but the display still shows 'X' and 'Y'. And the Y axis do not respond anymore to jog commands (that should be going to the 3rd axis, that do not exist - so nothing is done).
I think the correct should be the display to show axis 'X' an 'Z'. Is this correct?

**Reply by Paciente8159** [reply]
The representations is in UGS. I don't know if I can directly control that. I may take a look at UGS if there is a way to do that. The GRBL protocol does not name the axis anywhere if I recall correctly.

**Reply by mvackel** [reply]
Well, I'm referring to the display. It still shows 'X' and 'Y'.  Couldn't get it to show 'X' and 'Z' and use these motors.

**Reply by Paciente8159** [reply]
Ahhh. 
I think when I made the graphic display module I did not considered that case. And the changes you are referring are all at the graphic module level (not at core code).
The modification I showed above work on the parser level, so that the parser actually recognizes Z commands and not Y for 2-axis machine.

I will need to look at the graphic card module to see what would be needed to make that kind of changes.
